File: 2043-simple-bank-system.java
class SimpleBankSystem {
    long[] balance;

    public SimpleBankSystem(long[] balance) {
        this.balance = balance;
    }

    public boolean transfer(int account1, int account2, long money) {
        if (!isValid(account1) || !isValid(account2)) {
            return false;
        }
        if ((balance[account1 - 1] - money) >= 0) {
            balance[account1 - 1] -= money;
            balance[account2 - 1] += money;
            return true;
        }
        return false;
    }

    public boolean deposit(int account, long money) {
        if (!isValid(account)) {
            return false;
        }
        balance[account - 1] += money;
        return true;
    }

    public boolean withdraw(int account, long money) {
        if (!isValid(account)) {
            return false;
        }
        if (balance[account - 1] - money < 0) {
            return false;
        }
        balance[account - 1] -= money;
        return true;
    }

    private boolean isValid(int account) {
        return account > 0 && account <= balance.length;
    }

    public static void main(String[] args) {
        SimpleBankSystem bank = new SimpleBankSystem(new long[] { 10, 100, 20, 50, 30 });
        System.out.println(bank.withdraw(3, 10) + ", expected: true");
        System.out.println(bank.transfer(5, 1, 20) + ", expected: true");
        System.out.println(bank.deposit(5, 20) + ", expected: true");
        System.out.println(bank.transfer(3, 4, 15) + ", expected: false");
        System.out.println(bank.withdraw(10, 50) + ", expected: false");
    }
}

File: 2043-simple-bank-system.py
from typing import List


class Bank:

    def __init__(self, balance: List[int]):
        self.balance = balance

    def transfer(self, account1: int, account2: int, money: int) -> bool:
        if not self.is_valid(account1) or not self.is_valid(account2):
            return False
        account1 -= 1
        account2 -= 1

        account1_after = self.balance[account1] - money
        if account1_after >= 0:
            self.balance[account1] = account1_after
            self.balance[account2] += money
            return True

        return False

    def deposit(self, account: int, money: int) -> bool:
        if not self.is_valid(account):
            return False
        account -= 1

        self.balance[account] += money
        return True

    def withdraw(self, account: int, money: int) -> bool:
        if not self.is_valid(account):
            return False
        account -= 1

        if self.balance[account] - money < 0:
            return False

        self.balance[account] -= money
        return True

    def is_valid(self, account: int) -> bool:
        return account > 0 and account <= len(self.balance)


obj = Bank([10, 100, 20, 50, 30])
print(f'{obj.withdraw(3, 10)}, expected: True')
print(f'{obj.transfer(5, 1, 20)}, expected: True')
print(f'{obj.deposit(5, 20)}, expected: True')
print(f'{obj.transfer(3, 4, 15)}, expected: False')
print(f'{obj.withdraw(10, 50)}, expected: False')
